packages/firestore/src/platform/browser_lite/fetch_connection.ts

```diff
@@ -69,9 +69,11 @@ export class FetchConnection extends RestConnection {
     }
 
     if (!response.ok) {
+      const responseBody = await response.json();
       throw new FirestoreError(
         mapCodeFromHttpStatus(response.status),
-        'Request failed with error: ' + response.statusText
+        //response body is an array with error object
+        'Request failed with error: ' + responseBody[0].error.message
       );
```

Comment:
statusText here is only "Bad request". 
Backend rejected the request, but in this file, the this.fetchImpl didn't fail. It passed through the try catch block, and error message send by backend is stored inside the resposne.json.

Wonder
1.  if there is a way to make the fetchImp fail if the backend rejected the request, so that we can get the error message inside the catch block.
2. if we can't , can we get the error message in some other ways, not from response.json(). Does the Response type include fields sth like error, or statusMessage 🤔

Reply:
This is basically exactly what webchannel_connection.ts does, so I think your solution is quite reasonable.

https://github.com/firebase/firebase-js-sdk/blob/941ad29660a386059e5618b514d43cc5d409e642/packages/firestore/src/platform/browser/webchannel_connection.ts#L90-L131